lineman/app/services/ability_service.coffee

```diff
@@ -68,3 +68,19 @@ angular.module('loomioApp').factory 'AbilityService', (CurrentUser) ->
 
     canManageMembershipRequests: (group) ->
       (group.membersCanAddMembers and CurrentUser.isMemberOf(group)) or @canAdministerGroup(group)
+
+    canViewGroup: (group) ->
+      group.visibleToPublic() or
+      CurrentUser.isMemberOf(group) or
+      (group.visibleToOrganisation() and CurrentUser.isMemberOf(group.parent()))
+
+    canJoinGroup: (group) ->
+      (group.membershipGrantedUpon == 'request') and
+      @canViewGroup(group) and
+      !CurrentUser.isMemberOf(group)
+
+    canRequestMembership: (group) ->
```

Comment:
I think we need to catch up and write some specs for this file. Maybe even consider breaking it up into sections? @gdpelican